### PR TITLE
Implements facility and routes endpoints

### LINKED ISSUE #39: Amenities
@amalaabraham

### PR DESCRIPTION
**Pull request by arihant-2310**
Subject: "<feat #39 >"
Assignees: "Kites-hackathon-squad"

# Closes/Fixes/Resolves

### Closes #39 

# Explain the feature/fix

<p>This pull request adds certain end points for creation, updation and deletion of facilities and facility types along with the end
point for creation of routes. <p>

## Screenshots

<details>
<summary>View Screenshots</summary>

image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- [ ] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: src/facilities/dto/create-facility.dto.ts [hunk 1]
```diff
@@ -1 +1,45 @@
...
+  description: string;
+
+  @ApiProperty({ type: 'string' })
+  latitude: string;
+
+  @ApiProperty({ type: 'string' })
+  longitude: string;
+
+  @ApiProperty({ type: 'string' })
+  contact: string;
```
Comment (riginoommen) [inline src/facilities/dto/create-facility.dto.ts]: What is planning to store here.? can you please give a general view on this

File: src/facilities/dto/update-facility.dto.ts [hunk 2]
```diff
@@ -1,4 +1,36 @@
...
+  description: string;
+
+  @ApiProperty({ type: 'string' })
+  latitude: string;
+
+  @ApiProperty({ type: 'string' })
+  longitude: string;
+
+  @ApiProperty({ type: 'string' })
+  contact: string;
```
Comment (riginoommen) [inline src/facilities/dto/update-facility.dto.ts]: What is planning to store here.? can you please give a general view here.

File: src/facilities/facilities.controller.ts [hunk 3]
```diff
@@ -18,36 +20,53 @@ import { ApiTags } from '@nestjs/swagger';
...
-  @Get(':id')
-  findOne(@Param('id') id: string) {
-    return this.facilitiesService.findOne(+id);
+  @Get('type/:typeId')
+  findOne(@Param('typeId') id: number) {
+    return this.facilitiesService.findTypeById(+id);
   }
 
-  @Put(':id')
+  @Get('/:id')
```
Comment (riginoommen) [inline src/facilities/facilities.controller.ts:42]: Which data you are calling on this api.

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: src/facilities/dto/create-facility.dto.ts (hunk 1 above)
Comment (arihant-2310) [reply src/facilities/dto/create-facility.dto.ts]: To store the details about each facility like their address, photos, contact. Each entry belongs to a particular facility type. For eg. Theatres, Museums, View points are entries for a facility type "Entertainment", Hospitals, Medical Shops are part of facility type "Healthcare".

File: src/facilities/dto/update-facility.dto.ts (hunk 2 above)
Comment (arihant-2310) [reply src/facilities/dto/update-facility.dto.ts]: A data transfer object template which  would be required to update an already created facility.

File: src/facilities/facilities.controller.ts (hunk 3 above)
Comment (arihant-2310) [reply src/facilities/facilities.controller.ts:42]: There may be different types of facilities like Entertainment, Health, Finance etc . To get the details about the particular facility type we used this endpoint.

File: src/facilities/dto/create-facility.dto.ts (hunk 1 above)
Comment (sreenadh) [reply src/facilities/dto/create-facility.dto.ts]: @arihant-2310 what is planning to store in the field 'contact'? phone number or address ?

File: src/facilities/facilities.controller.ts (hunk 3 above)
Comment (sreenadh) [reply src/facilities/facilities.controller.ts:42]: @riginoommen we have field in facilities to store the type_id. And this API is part of CRUD of facility_types

File: src/facilities/migrations/1611948310353-createtabletype.ts [hunk 4]
```diff
@@ -4,7 +4,7 @@ export class createtabletype1611948310353 implements MigrationInterface {
   public async up(queryRunner: QueryRunner): Promise<void> {
     await queryRunner.createTable(
       new Table({
-        name: 'type',
+        name: 'facilityType',
```
Comment (sreenadh) [inline src/facilities/migrations/1611948310353-createtabletype.ts]:
```suggestion
        name: 'facility_types',
```

File: src/facilities/migrations/1611948310353-createtabletype.ts [hunk 5]
```diff
@@ -26,6 +26,6 @@ export class createtabletype1611948310353 implements MigrationInterface {
   }
 
   public async down(queryRunner: QueryRunner): Promise<void> {
-    await queryRunner.query(`DROP TABLE "type"`);
+    await queryRunner.query(`DROP TABLE "facilityType"`);
```
Comment (sreenadh) [inline src/facilities/migrations/1611948310353-createtabletype.ts]:
```suggestion
    await queryRunner.query(`DROP TABLE "facility_types"`);
```

File: src/facilities/migrations/1611981680527-facilities.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import { MigrationInterface, QueryRunner, Table } from 'typeorm';
+
+export class facilities1611981680527 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.createTable(
+      new Table({
+        name: 'facility',
```
Comment (sreenadh) [inline src/facilities/migrations/1611981680527-facilities.ts]:
```suggestion
        name: 'facilities',
```

File: src/facilities/migrations/1611981680527-facilities.ts [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            isNullable: true,
+          },
+          {
+            name: 'imageUrl',
+            type: 'varchar',
+            isNullable: true,
+          },
+          {
+            name: 'status',
+            type: 'varchar',
```
Comment (sreenadh) [inline src/facilities/migrations/1611981680527-facilities.ts]:
```suggestion
            type: 'number',
```

File: src/facilities/migrations/1611981680527-facilities.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+          },
+        ],
+      }),
+      true,
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`DROP TABLE "facility"`);
```
Comment (sreenadh) [inline src/facilities/migrations/1611981680527-facilities.ts]:
```suggestion
    await queryRunner.query(`DROP TABLE "facilities"`);
```

File: src/facilities/pipes/facility-status-validation.pipe.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import { CreateFacilityStatus } from './../create-facility-status.enum';
+
+export class TaskStatusValidationPipe implements PipeTransform {
+  readonly allowedStatuses = [
+    CreateFacilityStatus.OPEN,
+    CreateFacilityStatus.IN_ACTIVE,
+    CreateFacilityStatus.CLOSED,
+  ];
+  transform(value: any) {
+    value = value.toUpperCase();
```
Comment (sreenadh) [inline src/facilities/pipes/facility-status-validation.pipe.ts:11]: status is number

File: src/routes/dto/create-route.dto.ts [hunk 10]
```diff
@@ -1 +1,42 @@
...
+  @IsNotEmpty()
+  status: string;
+
+  @ApiProperty({ type: 'number' })
+  @IsNotEmpty()
+  totalDistance: number;
+
+  @ApiProperty({ type: 'number' })
+  @IsNotEmpty()
+  review: number;
```
Comment (sreenadh) [inline src/routes/dto/create-route.dto.ts]:
```suggestion
  rating: number;
```

File: src/routes/entities/route.entity.ts [hunk 11]
```diff
@@ -15,6 +15,9 @@ export class Route extends BaseEntity {
   @Column()
   routeName: string;
 
+  @Column()
+  review: number;
```
Comment (sreenadh) [inline src/routes/entities/route.entity.ts]:
```suggestion
  rating: number;
```

File: src/routes/migrations/1611943981900-routes.ts [hunk 12]
```diff
@@ -29,10 +29,18 @@ export class routes1611943981900 implements MigrationInterface {
             name: 'source',
             type: 'varchar',
           },
+          {
+            name: 'description',
+            type: 'varchar',
```
Comment (sreenadh) [inline src/routes/migrations/1611943981900-routes.ts]:
```suggestion
            type: 'text',
```

File: src/routes/migrations/1611943981900-routes.ts [hunk 13]
```diff
@@ -29,10 +29,18 @@ export class routes1611943981900 implements MigrationInterface {
...
+          },
           {
             name: 'destination',
             type: 'varchar',
           },
+          {
+            name: 'status',
+            type: 'varchar',
```
Comment (sreenadh) [inline src/routes/migrations/1611943981900-routes.ts]:
```suggestion
            type: 'number',
```

File: src/routes/routes.repository.ts [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+
+@EntityRepository(Route)
+export class routeRepository extends Repository<Route> {
+  async createRoute(createroutedto: CreateRouteDto): Promise<any> {
+    const {
+      routeName,
+      description,
+      imageUrl,
+      source,
+      review,
```
Comment (sreenadh) [inline src/routes/routes.repository.ts]:
```suggestion
      rating,
```

File: src/routes/routes.repository.ts [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+      status,
+    } = createroutedto;
+
+    const route = new Route();
+    route.description = description;
+    route.destination = destination;
+    route.routeName = routeName;
+    route.imageUrl = imageUrl;
+    route.source = source;
+    route.review = review;
```
Comment (sreenadh) [inline src/routes/routes.repository.ts]:
```suggestion
    route.rating = rating;
```

File: src/routes/dto/create-route.dto.ts [hunk 16]
```diff
@@ -1 +1,42 @@
...
+
+  @ApiProperty({ type: 'string' })
+  @IsString()
+  @IsNotEmpty()
+  destination: string;
+
+  @ApiProperty({ type: 'string' })
+  @IsString()
+  @IsNotEmpty()
+  status: string;
```
Comment (sreenadh) [inline src/routes/dto/create-route.dto.ts]:
```suggestion
  status: number;
```

**Review by sreenadh — CHANGES_REQUESTED** [review]
Please see the comments

**Pull request closed without merging**